app/src/main/java/com/gholem/moneylab/features/add/AddTransactionFragment.kt

```diff
@@ -52,11 +54,30 @@ class AddTransactionFragment : BaseFragment<FragmentAddBinding, AddTransactionVi
         viewModel.navigation.observe(this, navigation::navigate)
     }
 
+    //only: fragment back to fragment\\ savedStateHandle
     private fun observeCategoryChange() {
         findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<Int>(KEY_CATEGORY)
             ?.observe(viewLifecycleOwner) { result -> dataAdapter.setCategory(result) }
     }
 
+    private fun showDateDialog(_position: Int) {
+        viewModel.getDateCalendar()
+        position = _position
+
+        val rightNow: Calendar = Calendar.getInstance()
+
+        var contextOfActivity = this.context
+        if (contextOfActivity != null) {
+            DatePickerDialog(
+                contextOfActivity,
```

Comment:
This is fine, but there is a better method to deal with nullable objects as I said on the call:

```
this.context?.let { context ->
      // use *context* here
}
```
But in that case there is even better way - Fragment has a method `requireContext()` which you can use here instead of manually dealing with the nullable context, so use it please instead of `contextOfActivity`